redisearch/client.py

```diff
@@ -102,9 +108,17 @@ class IndexDefinition(object):
     SCORE = 'SCORE'
     PAYLOAD_FIELD = 'PAYLOAD_FIELD'
         
-    def __init__(self, prefix=[], filter=None, language_field=None, language=None, score_field=None, score=1.0, payload_field=None):
-
-        args = [self.ON, self.HASH]
+    def __init__(self, prefix=[], filter=None, language_field=None, language=None, score_field=None, score=1.0, payload_field=None, index_type=None):
+        args = []
+
+        if index_type is IndexType.HASH:
+            args.append(self.ON)
+            args.append(self.HASH)
```

Comment:
Since these variables aren't reused, I'd just pass in the strings during creation. The variables don't belong in the class scope *unless* you're using them elsewhere.

If you're using them elsewhere and not modifying them, then they properly should be properties.

@property
def hash(self):
    return "HASH"

Reply:
I want to be consistent with all the constant strings (L100-109)

Reply:
Right. Instead of having the variable at the class level, since they're immutable, they should really be properties.